### PR TITLE
refactor(dot): Microtype name

### PR DESCRIPTION
**Pull request by jesperolsson-se**

### CODE REVIEW

File: src/main/java/org/schemaspy/output/dot/schemaspy/name/Null.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package org.schemaspy.output.dot.schemaspy.name;
+
+/**
+ * Null object for name.
+ */
+public final class Null implements Name {
```
Comment (npetzall) [inline src/main/java/org/schemaspy/output/dot/schemaspy/name/Null.java:6]: Null is a loaded word, and it's actually an EmptyName. Could we rename it to EmptyName?

**Review by npetzall — CHANGES_REQUESTED** [review]

File: src/main/java/org/schemaspy/output/dot/schemaspy/name/Null.java (hunk 1 above)
Comment (jesperolsson-se) [reply src/main/java/org/schemaspy/output/dot/schemaspy/name/Null.java:6]: Would `Empty` suffice ("name" can be inferred from the scope of the package and/or interface)?

Comment (npetzall) [reply src/main/java/org/schemaspy/output/dot/schemaspy/name/Null.java:6]:
```
Name diagramName = new Degree(            
  twoDegreesOfSeparation,            
  new Implied(               
     includeImplied,                
     new Empty()           
  )        
);
```

Compilers, IDEs are better at inferring than humans. Empty is generic term so I think it hurts readability/understandability.

Import statement is at the top of the file.

File: src/main/java/org/schemaspy/output/dot/schemaspy/DotTableFormatter.java [hunk 2]
```diff
@@ -80,8 +84,15 @@ public DotTableFormatter(
 
         DotConnectorFinder finder = new DotConnectorFinder();
 
-        String diagramName = (twoDegreesOfSeparation ? "twoDegreesRelationshipsDiagram" : "oneDegreeRelationshipsDiagram") + (includeImplied ? "Implied" : "");
-        dotFormat.writeHeader(diagramName, true, dot);
+        Name diagramName = new Degree(
+            twoDegreesOfSeparation,
+            new Implied(
+                includeImplied,
+                new Simple("")
```
Comment (npetzall) [inline src/main/java/org/schemaspy/output/dot/schemaspy/DotTableFormatter.java:91]: Replace with Empty Name?

**Review by npetzall — APPROVED** [review]

**Pull request merged**